packages/ts/src/components/donut/index.ts

```diff
@@ -57,24 +57,19 @@ export class Donut<Datum> extends ComponentCore<Datum[], DonutConfig<Datum>, Don
 
   _render (customDuration?: number): void {
     const { config, datamodel, bleed } = this
-    const data = datamodel.data
+    const data = config.showEmptySegments ? datamodel.data : datamodel.data.filter((d, i) => getNumber(d, config.value, i))
```

Comment:
@reb-dev I should have not committed this line, it might have confused you, sorry. That's why the colors get messed up. Let's change it back to just `data = datamodel.data`

Reply:
@reb-dev In that case the colors will be correct, but if the donut has only one non-empty segment, where'll be a confusing gap at start:
<img width="599" alt="image" src="https://user-images.githubusercontent.com/755708/214375216-1bb1c72f-1655-4e1c-be1b-7251ff4049d9.png">

Reply:
@rokotyan Fixed! I actually like the idea of filtering the data, I tweaked it a bit so that the indices are preserved. Let me know if it's ok?